Standardize FontAwesome usage

dannon opened the pull request:
Replace the string-based FontAwesome library approach with direct SVG icon imports for better tree-shaking and type safety.

- Better tree-shaking (only used icons get bundled)
- Type safety improvements
- Reduced global state pollution
- More explicit dependencies


Changes: 

Removed library.add() calls: Eliminated all library.add() statements that were registering icons globally
Removed library imports: Removed import { library } from "@fortawesome/fontawesome-svg-core"
Updated icon usage patterns: Changed from string-based icon references to direct icon object references


Before:
```
  import { library } from "@fortawesome/fontawesome-svg-core";
  import { faCopy } from "@fortawesome/free-regular-svg-icons";
  library.add(faCopy);

  // Template usage:
  <FontAwesomeIcon icon="copy" />
  // or
  <FontAwesomeIcon :icon="['far', 'copy']" />
```

After:
```
  import { faCopy } from "@fortawesome/free-regular-svg-icons";

  // Template usage:
  <FontAwesomeIcon :icon="faCopy" />
```

The icon plugin/mixin are still up in the air.  I could see pushing it *all* through a `GIcon` if we wanted, instead of FontAwesome directly, and can make that changes.  This first standardization pass is important though, either way, just to get it all aligned.

ElectronicBlueberry commented on client/src/components/plugins/icons.js:
With removing the library function, I would be in favor of also removing this file. I feel like it's an ambiguous abstraction, that makes it harder to trace the actual icon. But maybe there's an upside I'm missing

dannon replied:
@ElectronicBlueberry That's what I was thinking, too, 👍 

(mentioned above, the only question is on whether we want a `GIcon` or something as a compatibility interface between Galaxy and Icons in case we need/want to change down the road.  But since we're locally importing a slew of direct SVGs from fontawesome anyway, I don't see a benefit here.)

dannon committed:
Remove library.add() calls from Vue components

Updated components to import icon objects directly instead of using library.add(). Components now bind icons with :icon="faIconName" rather than icon="fa-icon-name".

dannon committed:
Update icon utility files to remove library.add() calls

Updated Workflow and Libraries icon files to export icon arrays instead of calling library.add(). Also updated useInvocationGraph to remove library registration.

dannon committed:
Update documentation for icon usage

Changed icon prop documentation to reflect using icon objects instead of library.add().

dannon committed:
Remove icon plugin

Deleted the icon plugin file and removed it from the plugin index, mountVueComponent, and test helpers. Some tests will fail temporarily because a few components still use string-based icon references that relied on the global icon library.

dannon committed:
Add Galaxy icon generation infrastructure

Created a script to auto-generate TypeScript exports from the Galaxy icon JSON. The gulpfile now runs this during the icon build step, creating galaxyIcons.ts with properly typed IconDefinition exports.

dannon committed:
Update AboutGalaxy to use Galaxy logo icon object

Imported galaxyLogo from the generated Galaxy icons and used it directly instead of the string array format.

dannon committed:
Add IconLike union type for mixed icon usage

Update Heading, MastheadItem, and icon generation to accept both FontAwesome and Galaxy icons through the new IconLike type.

dannon committed:
Fix ts icon generation formatting

Improved icon generation to use manual formatting instead of JSON.stringify for better readability and consistency with other icon files.

dannon committed:
Fix remaining icon usage and imports

Updated various components to use direct icon imports instead of string-based references. Removed remaining library.add() calls and updated templates to bind icon objects.

dannon committed:
Replace Icon component with FontAwesomeIcon (partial)

Updated DataDialog, GridBoolean, GridList, GridPage, and Masthead to use FontAwesomeIcon instead of the custom Icon component.

dannon committed:
Replace remaining Icon components with FontAwesomeIcon

dannon committed:
Fix typing in MastheadItem for optional url prop

dannon committed:
Fix tests after icon plugin removal

dannon committed:
Convert window manager icon to use FontAwesome icon object

dannon committed:
Convert all remaining icon string references to use icon objects

dannon committed:
Fix remaining FontAwesomeIcon string references in Libraries and ToolPanel components

dannon committed:
Remove library.add() completely - use direct icon object imports for Galaxy custom icons

dannon committed:
Yet more icon string swaps

dannon committed:
Fix remaining icon string references in notifications and library components

dannon committed:
Convert FormElement, FormDisplay, and FormInputs icon defaults to use icon objects

dannon committed:
Use icon objects for form icon props

dannon committed:
Fix formatting in build_icons.js to not conflict w/ prettier when generating docs

dannon committed:
Update template and generated file paths in TypeScript icon files for
path consistency

dannon committed:
Fix library folder checkbox icon format
